cis194/week4/weining/src/Exercise1.hs

```diff
@@ -0,0 +1,34 @@
+module Exercise1
+    ( fun1Original
+    , fun1
+    , fun2Original
+    , fun2
+    ) where
+
+import           Data.Bool
+
+fun1Original :: [Integer] -> Integer
+fun1Original [] = 1
+fun1Original (x:xs)
+    | even x = (x - 2) * fun1Original xs
+    | otherwise = fun1Original xs
+
+fun1 :: [Integer] -> Integer
+fun1 = (foldr (*) 1) .
+       (map (bool <$> (const 1) <*> (subtract 2) <*> even))
```

Comment:
In Haskell we often align symbols. 
```Haskell
fun1 = foldr (*) 1 -- since function application has the highest precedence, we don't need brackets around this.
     . map calc
   where
     calc = bool <$> const 1 <*> subtract 2 <*> even -- we are trying to get rid of some brackets here to reduce noise in the code base
```

Reply:
Nice style!